### PR TITLE
Fix(builder_unlock):Audit fixes

### PR DESCRIPTION
**Pull request by maksk7**

### CODE REVIEW

File: contracts/builder_unlock/src/contract.rs [hunk 1]
```diff
@@ -423,7 +437,9 @@ mod helpers {
             Uint128::zero()
         }
         // Tokens unlock linearly between start time and end time
-        else if timestamp < schedule.start_time + schedule.duration {
+        else if (timestamp < schedule.start_time + schedule.cliff + schedule.duration)
```
Comment (ShadoySV) [inline contracts/builder_unlock/src/contract.rs:440]: Let's move `+ schedule.cliff` to 436 line's right expression and the same to numerator of 443 line, to be consistent with the description of the Schedule structure. And then we can delete the if block starting from 458 line

File: contracts/builder_unlock/src/contract.rs [hunk 2]
```diff
@@ -303,12 +303,26 @@ fn execute_claim_receiver(
                 // Transfers Allocation Parameters ::
                 // 1. Save the allocation for the new receiver
                 alloc_params.proposed_receiver = None;
+
+                if let Some(sender_params) = PARAMS.may_load(deps.storage, &info.sender)? {
+                    alloc_params.amount = alloc_params.amount.checked_add(sender_params.amount)?;
+                }
```
[the remark this answers is not in the document]
Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs:309]: Fixed, thanks

File: contracts/builder_unlock/src/contract.rs (hunk 1 above)
Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs:440]: Fixed, thanks

File: contracts/builder_unlock/src/contract.rs [hunk 3]
```diff
@@ -300,14 +301,18 @@ fn execute_claim_receiver(
     match alloc_params.proposed_receiver {
         Some(proposed_receiver) => {
             if proposed_receiver == info.sender {
+                if let Some(sender_params) = PARAMS.may_load(deps.storage, &info.sender)? {
+                    return Err(StdError::generic_err(format!(
+                        "The proposed receiver already has an ASTRO allocation of {} ASTRO, that ends at {}",
+                        sender_params.amount,
+                        sender_params.unlock_schedule.start_time + sender_params.unlock_schedule.duration
```
Comment (ShadoySV) [inline contracts/builder_unlock/src/contract.rs:308]: sender_params.cliff should be added too

Comment (maksk7) [reply contracts/builder_unlock/src/contract.rs]: Ok, done

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**